poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java

```diff
@@ -96,4 +231,119 @@ public EscherDggRecord getDgg() {
+            blipRecord = new EscherMetafileBlip();
+            escherTag = 0;
+        } else {
+            blipRecord = new EscherBitmapBlip();
+            escherTag = (short) 0xFF;
+        }
+
+        blipRecord.setRecordId((short) (EscherBlipRecord.RECORD_ID_START + format));
+        switch (format) {
+            case PICTURE_TYPE_EMF:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_EMF);
+                break;
+            case PICTURE_TYPE_WMF:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_WMF);
+                break;
+            case PICTURE_TYPE_PICT:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_PICT);
+                break;
+            case PICTURE_TYPE_PNG:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_PNG);
+                break;
+            case PICTURE_TYPE_JPEG:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_JPEG);
+                break;
+            case PICTURE_TYPE_DIB:
+                blipRecord.setOptions(HSSFPictureData.MSOBI_DIB);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected picture format: " + format);
+        }
+
+        EscherBSERecord r = new EscherBSERecord();
+        r.setOptions((short) (0x0002 | (format << 4)));
+        r.setBlipTypeMacOS((byte) format);
+        r.setBlipTypeWin32((byte) format);
+        r.setTag(escherTag);
+        r.setBlipRecord(blipRecord);
+
+        addBSERecord(r);
+        return new HSSFPictureData(r);
+    }
+
+    /**
+     * Returns the pictures in the workbook associated with this {@link DrawingManager2}.
+     * <p>
+     * The pictures are ordered as they are stored in the worksheet.
+     *
+     * @return Pictures in the workbook.
+     */
+    @Internal
+    public List<HSSFPictureData> getAllPictures() {
+        ArrayList<HSSFPictureData> pictures = new ArrayList<>(bStoreContainer.getChildCount());
+        for (EscherRecord record : bStoreContainer) {
+            EscherBSERecord bse = (EscherBSERecord) record;
+
+            // We sometimes encounter sheets that have invalid BSE records
+            EscherBlipRecord blip = bse.getBlipRecord();
+            if (blip != null) {
+                pictures.add(new HSSFPictureData(bse));
+            } else {
+                LOGGER.atDebug().log("Encountered BSE record without a BLIP. BSE records must have a BLIP according to the specification.");
+            }
+        }
+        return pictures;
+    }
+
+    /**
+     * Returns the number of pictures in the workbook associated with this {@link DrawingManager2}.
+     *
+     * @return Number of pictures in the workbook.
+     */
+    @Internal
+    public int getPictureCount() {
+        return bStoreContainer.getChildCount();
+    }
+
+    /**
+     * Appends the given record to the {@link #bStoreContainer}.
+     */
+    private void addBSERecord(EscherBSERecord bse) {
```

Comment:
not a fan of the variable name here, escherRecord or escherBSERecord seems better